secrets/provider/vault/config.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2022 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package vault
+
+import (
+	"net/url"
+
+	"github.com/juju/errors"
+	"github.com/juju/schema"
+	"gopkg.in/juju/environschema.v1"
+
+	coreconfig "github.com/juju/juju/core/config"
+	"github.com/juju/juju/secrets/provider"
+)
+
+const (
+	EndpointKey      = "endpoint"
+	NamespaceKey     = "namespace"
+	TokenKey         = "token"
+	CACertKey        = "ca-cert"
+	ClientCertKey    = "client-cert"
+	ClientKeyKey     = "client-key"
+	TLSServerNameKey = "tls-server-name"
+	UnsealKeysKey    = "unseal-keys"
+)
+
+var configSchema = environschema.Fields{
+	EndpointKey: {
+		Description: "The vault service endpoint.",
+		Type:        environschema.Tstring,
+		Immutable:   true,
+		Mandatory:   true,
+	},
+	TokenKey: {
+		Description: "The vault access token.",
+		Type:        environschema.Tstring,
+		Secret:      true,
+	},
+	NamespaceKey: {
+		Description: "The namespace in which to store secrets.",
+		Type:        environschema.Tstring,
+	},
+	CACertKey: {
+		Description: "The vault CA certificate.",
+		Type:        environschema.Tstring,
+	},
+	ClientCertKey: {
+		Description: "The vault client certificate.",
+		Type:        environschema.Tstring,
+	},
+	ClientKeyKey: {
+		Description: "The vault client certificate key.",
+		Type:        environschema.Tstring,
+		Secret:      true,
+	},
+	UnsealKeysKey: {
```

Comment:
It's my belief that we should not be supporting the storage of Vault UnsealKeys in Juju at all. This is the fundamental encryption layer of vault and Juju is not in any position to store this information. Even the hint that Juju supports this has the potential to encourage users to take on practices that are not safe or implies that Juju can be a trusted source for this information.

For more context https://developer.hashicorp.com/vault/docs/concepts/seal

If this is to aid in development where a normal `vault -dev` mode will not suffice then we should build adequate tooling and scripts into our build pipeline to help us but including this in a production release stinks on Juju's part.

We try to promote our view on best practice for modeling the world and how to manage operations. I don't think this fits that narrative except to make a developers life easier for 2 seconds.

Reply:
they were just there for developers, removed now